sdk/core/azure_core_macros/src/tracing_new.rs

```diff
@@ -219,58 +219,34 @@ pub fn parse_new(attr: TokenStream, item: TokenStream) -> Result<TokenStream> {
 fn is_arc_of_self(path: &syn::Path) -> std::result::Result<(), String> {
     let segment = path.segments.last().unwrap();
     if segment.ident != "Arc" {
-        error!(
-            "Invalid return type for new function: Arc must be the first segment, found {:?}",
-            segment.ident
-        );
-        return Err(
-            "Invalid return type for new function: Arc must be the first segment".to_string(),
-        );
+        warn!("Arc must be the first segment, found {:?}", segment.ident);
+        return Err("Arc must be the first segment".to_string());
```

Comment:
Why are you tracing the same thing you're returning in an error. That shouldn't be necessary. The error will be returned to the compiler. And all these traces aren't free and will impact development performance.

Reply:
You could put that context into the errors. Anything that helps describe why the error happened and how to fix it seems like something any developer should see - not just those that enabled tracing.